CMSgov/design-system
# [NO-TICKET] Introduce `scan-dependents` sub-commands and implement code search

## Summary

This adds a wishlist item from https://github.com/CMSgov/design-system/pull/1966 and reorganizes the `scan-dependents` command to have sub-commands.

### Changed

- New commands:
  - `yarn scan-dependents versions` - was previously `yarn scan-dependents`
  - `yarn scan-dependents dependency <dependencyName>` - was previously `yarn scan-dependents --dependency <dependencyName>`
  - `yarn scan-dependents search <searchString>` - a new feature
  - Run `yarn scan-dependents --help` to see commands and options

## How to test

See the readme and `yarn scan-dependents --help`. Try some common formulas in the readme. I can't test anymore because I ran it _a lot_ and have reached my API rate limit 😂 

Something that might be nice in the future if we use this a lot is local caches of the repo lists (result of the `getDependentPackageData` function) saved by day in `scripts/scanDependents/cache` so we can skip 1-4 API calls for every search executed
